lib/puppet/node/environment.rb

```diff
@@ -362,7 +364,9 @@ def modules_by_path
     modulepath.each do |path|
       if Puppet::FileSystem.exist?(path)
         Dir.chdir(path) do
```

Comment:
![](https://memegenerator.net/img/instances/400x/80501394/but-y-tho.jpg)

Reply:
Looks like that's left over from fecf5d693dde7b9101d6f51eddf2a343afcbbcef when it used to `Dir.chdir` followed by `Dir.glob('*')`. The latter was removed in 6d499bde4ca11571a1878c7cc917e490c00e8823, but not the former. We should be able to remove that now. I'll update that.

Reply:
Actually there's a ticket about that https://tickets.puppetlabs.com/browse/PUP-9997, I'll handle that in a separate PR

Reply:
Submitted as https://github.com/puppetlabs/puppet/pull/8687